wicket-core/src/main/java/org/apache/wicket/Session.java

```diff
@@ -915,6 +924,10 @@ public int nextPageId()
 	 */
 	public final IPageManager getPageManager()
 	{
+		if (Boolean.TRUE.equals(RequestCycle.get().getMetaData(PAGES_UNLOCKED))) {
+			throw new WicketRuntimeException("pages have already been unlocked - synchronized access is no longer possible");
```

Comment:
I am not sure how understandable this message would be for the average developer. Maybe something like "The request [cycle] has been processed. Access to the pages is no more allowed" ?